x-pack/plugins/actions/server/builtin_action_types/servicenow/types.ts

```diff
@@ -253,6 +253,7 @@ export interface SNProductsConfigValue {
   useImportAPI: boolean;
   importSetTable: string;
   commentFieldKey: string;
+  appId?: string;
```

Comment:
Wonder why this is optional...

Reply:
I make it optional because ITOM does not have an application.